libs/sdk-ui/src/execution/useInsightDataView.ts

```diff
@@ -0,0 +1,101 @@
+// (C) 2019-2021 GoodData Corporation
+import { IAnalyticalBackend, IPreparedExecution } from "@gooddata/sdk-backend-spi";
+import { INullableFilter, ObjRef } from "@gooddata/sdk-model";
+import identity from "lodash/identity";
+import {
+    useBackendStrict,
+    useWorkspaceStrict,
+    resolveUseCancelablePromisesStatus,
+    resolveUseCancelablePromisesError,
+    UseCancelablePromiseState,
+    DataViewFacade,
+    GoodDataSdkError,
+} from "../base";
+import { useExecutionDataView } from "./useExecutionDataView";
+import { DataViewWindow } from "./withExecutionLoading";
+import { useInsight } from "./useInsight";
+
+/**
+ * @beta
+ */
+export interface IUseInsightDataViewConfig {
+    /**
+     * Reference to the insight for which you want to get the data view.
+     *
+     * Note: When the reference or identifier is not provided, hook is locked in a "pending" state.
+     */
+    insight?: ObjRef;
+
+    /**
+     * Optionally transform prepared insight execution, before it's executed (e.g. modify dimensions, sorts or date formatting).
+     */
+    transformExecution?: (insightExecution: IPreparedExecution) => IPreparedExecution;
+
+    /**
+     * Optionally specify filters to merge with filters already defined in the insight.
+     */
+    filters?: INullableFilter[];
+
+    /**
+     * Optionally, you can define only a specific "window" of data to load.
+     * This is useful if you want to page data.
+     */
+    window?: DataViewWindow;
+
+    /**
+     * Backend to work with.
+     *
+     * Note: the backend must come either from this property or from BackendContext. If you do not specify
+     * backend here, then the executor MUST be rendered within an existing BackendContext.
+     */
+    backend?: IAnalyticalBackend;
+
+    /**
+     * Workspace where execution should be executed.
+     *
+     * Note: the workspace must come either from this property or from WorkspaceContext. If you do not specify
+     * workspace here, then the executor MUST be rendered within an existing WorkspaceContext.
+     */
+    workspace?: string;
+}
+
+/**
+ * React hook to get data for a specific insight.
+ *
+ * @beta
+ */
+export function useInsightDataView(
+    config: IUseInsightDataViewConfig,
+    deps?: React.DependencyList,
+): UseCancelablePromiseState<DataViewFacade, GoodDataSdkError> {
+    const { insight: insightRef, transformExecution = identity, filters, window } = config;
+    const backend = useBackendStrict(config.backend);
```

Comment:
Idea: we could maybe add a second optional parameter to the strict hooks, something like `context` which could make the error messages more useful (the `context` would be used in the error message, so that the caller does not have to make the whole error message).